.github/workflows/docs.yaml

```
@@ -0,0 +1,60 @@
+on:
+  push:
+    branches:
+      - master
+    paths:
+      - .github/workflows/docs.yaml
+      - docs/**
+
+name: Deploy DataFusion site
+
+jobs:
+  build-docs:
+    name: Build docs
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout docs sources
+        uses: actions/checkout@v3
+
+      - name: Checkout asf-site branch
+        uses: actions/checkout@v3
+        with:
+          ref: asf-site
+          path: asf-site
+
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+
+      - name: Install dependencies
+        run: |
+          set -x
+          python3 -m venv venv
+          source venv/bin/activate
+          pip install -r docs/requirements.txt
+
+      - name: Build docs
+        run: |
+          set -x
+          source venv/bin/activate
+          cd docs
+          ./build.sh
+
+      - name: Copy & push the generated HTML
+        run: |
+          set -x
+          cd asf-site/
+          # delete anything but: 1) '.'; 2) '..'; 3) .git/
+          rm -rf .github .buildinfo .gitattributes .github_changelog_generator .gitignore .gitmodules
+          find ./ | grep -vE "^./$|^../$|^./.git" | xargs rm -rf
+          cp ../.asf.yaml .
+          cp -rT ../docs/build/html/ .
```

FYI: We can use `rsync` for this like https://github.com/apache/arrow-site/blob/main/.github/workflows/deploy.yml#L98-L107 :

```suggestion
          rsync \
            -a \
            --delete \
            --exclude '/.git/' \
            ../docs/build/html/ \
            ./
          cp ../.asf.yaml .
```